vic/drivers/shared_image/src/vic_init_output.c

```diff
@@ -24,270 +24,319 @@
-                 -global_param.nrecs);
+        initialize_save_data(&(all_vars[i]), &(atmos[i]), &(soil_con[i]),
+                             veg_con[i], veg_lib[i], &lake_con, out_data[i],
+                             &(save_data[i]));
     }
 
     if (mpi_rank == 0) {
-        // determine which variables will be written to the history file
-        parse_output_info(filep.globalparam, out_data);
+        // count the number of streams and variables in the global parameter file
+        count_nstreams_nvars(filep.globalparam, &(options.Noutstreams),
+                             nstream_vars);
+
+        // If there weren't any output streams specified, get the defaults
+        if (options.Noutstreams == 0) {
+            default_outputs = true;
+            get_default_nstreams_nvars(&(options.Noutstreams), nstream_vars);
+        }
+    }
 
-        // open the netcdf history file
-        initialize_history_file(&nc_hist_file);
+    // broadcast Noutstreams and nstream_vars
+    status = MPI_Bcast(&(options.Noutstreams), 1, MPI_AINT, 0, MPI_COMM_VIC);
+    if (status != MPI_SUCCESS) {
+        log_err("MPI error %d\n", status);
+    }
+    status = MPI_Bcast(&(nstream_vars), MAX_OUTPUT_STREAMS, MPI_AINT, 0,
+                       MPI_COMM_VIC);
+    if (status != MPI_SUCCESS) {
+        log_err("MPI error %d\n", status);
     }
 
-    // broadcast which variables to write.
-    for (i = 0; i < N_OUTVAR_TYPES; i++) {
-        status = MPI_Bcast(&out_data[0][i].write, 1, MPI_C_BOOL,
-                           0, MPI_COMM_VIC);
-        if (status != MPI_SUCCESS) {
-            log_err("MPI error in vic_init_output(): %d\n", status);
-        }
+    // allocate output streams
+    output_streams = calloc(options.Noutstreams, sizeof(*output_streams));
+    if (output_streams == NULL) {
+        log_err("Memory allocation error");
+    }
+    // allocate netcdf history files array
+    nc_hist_files = calloc(options.Noutstreams, sizeof(*nc_hist_files));
+    if (nc_hist_files == NULL) {
+        log_err("Memory allocation error");
     }
 
-    // broadcast history file info. Only the master process will write to it,
-    // but the slave processes need some of the information to initialize as
-    // well (particularly which variables to write and dimension sizes)
-    status = MPI_Bcast(&nc_hist_file, 1, mpi_nc_file_struct_type,
-                       0, MPI_COMM_VIC);
-    if (status != MPI_SUCCESS) {
-        log_err("MPI error in vic_init_output(): %d\n", status);
+    // allocate memory for streams, initialize to default/missing values
+    for (streamnum = 0; streamnum < options.Noutstreams; streamnum++) {
+        setup_stream(&(output_streams[streamnum]), nstream_vars[streamnum],
+                     local_domain.ncells_active);
     }
 
-    // initialize netcdf info for output variables
-    vic_nc_info(&nc_hist_file, out_data, nc_vars);
+    if (mpi_rank == 0) {
+        if (!default_outputs) {
```

Comment:
I would swap the statements here. Typically `if` statements are easier to read if they are affirmative rather than negative. So, I would suggest:
```C
           // determine which variables will be written to the history file
         if (default_outputs) {
             set_output_defaults(&output_streams, dmy_current, NETCDF4_CLASSIC);
         }
         else {
             // set output defaults
            parse_output_info(filep.globalparam, &output_streams, dmy_current);
         }
     }
 
```